### PR TITLE
Fix msal test app after merging with the fadi/release/5.4.0

### PR DESCRIPTION
**Pull request by Yuki-YuXin**
Fix update according to the following team feedback:
**Batch2**
- [hero scenario 6, use case 2.2.1] Use email and OTP to get token and sign in
getParcelable(Constants.STATE) 
- [use case 2.2.6] Ability to provide scope to control auth strength of the token
Update the scope of profile to User.Read 
Update the test app EmailSignInSignUp page: signIn(scopes = User.Read)
- [use case 1.2.2] User is not registered with given email
Update test app EmailPasswordSignInSignUp page: displayDialog("Unexpected result", actionResult.errorMessage) in sign in function
- [use case 1.2.6] Ability to provide scope to control auth strength of the token
Update the test app EmailPasswordSignInSignUp page: signIn(scopes = User.Read)
Update test app EmailPasswordSignInSignUp page: displayDialog("Unexpected result", actionResult.errorMessage) in sign in function
- [hero scenario 2, use case 2.1.2] Signup user with custom attributes with verify OTP as last step
Update the test app SignUpCode page: currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpCodeRequiredState)!!
- [hero scenario 2, use case 2.1.2] Signup user with custom attributes with verify OTP as last step
Update the test app SignUpCode page: currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpCodeRequiredState)!!
- [hero scenario 2, use case 2.1.2] Signup user with custom attributes with verify OTP as last step

- [hero scenario 3, use case 2.1.3] Verify email OTP first and then collect custom attributes
Update the test app SignUpAttributes page: currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpAttributesRequiredState)!!
SignUpAttributesUI disorder

- [use case 1.1.4] Verify email address using email OTP and then set password 
In SignUpCodeFragment modify to currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpCodeRequiredState)!!
In SignUpPasswordFragment modify to currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpPasswordRequiredState)!!

**Batch1**
- [use case 1.2.2] User is not registered with given email
No error is popped up in UI (Probably this is a bug of the test app)
- [use case 2.2.5] Resend email OTP
App crashing
- [use case 1.2.3] Password is incorrect
No error is popped up in UI (Probably this is a bug of the test app)
- [hero scenario 3, use case 2.1.3] Verify email OTP first and then collect custom attributes
When submitting attributes (had to hard-code some), the test application doesn't return to the sign in / sign up screen
- Authenticate user (get token) through web browser, retrieve tokens through native auth SDK interface (by setting forceRefresh=true): 
EmailSignInSignUpFragment does not have logic for handling redirects
 - [hero scenario 6, use case 2.2.1] Use email and OTP to get token and sign in

### CODE REVIEW

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt [hunk 1]
```diff
@@ -108,66 +114,90 @@ class EmailSignInSignUpFragment : Fragment() {
...
-                            codeLength =  actionResult.codeLength,
-                            sentTo = actionResult.sentTo,
-                            channel = actionResult.channel
+            val email = binding.emailText.text.toString()
+
+            System.out.println("1")
+            val actionResult = authClient.signIn(
+                username = email
+            )
+            System.out.println("2")
```
Comment (rmccahill) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt]: Looks like some left over debugging here, same on line 119

File: msal/versioning/version.properties [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #Wed Aug 01 15:24:11 PDT 2018
-versionName=5.3.1
+versionName=5.4.0
```
Comment (rmccahill) [inline msal/versioning/version.properties]: Should these versions be updated as part of these changes? Same applies for the changes in `msalautomationapp/build.gradle`, and `msal/build.gradle`

Comment (Yuki-YuXin) [reply msal/versioning/version.properties]: I think the changes is because of the merging with the dev or fadi/release/branch

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt (hunk 1 above)
Comment (Yuki-YuXin) [reply testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt]: Sure. I will remove this line. We may not need test app any longer if the E2E tests are done.

Comment (rmccahill) [reply testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt]: I think it will still be useful for manually debugging flows, and for checking incidents for on call engineers as well

**Review by rmccahill — APPROVED** [review]

File: changelog [hunk 3]
```diff
@@ -3,6 +3,10 @@ MSAL Wiki : https://github.com/AzureAD/microsoft-authentication-library-for-andr
 vNext
 ----------
 
+Version 5.4.1-RC1
+---------
+- [PATCH] Update common @17.6.0-RC1
```
Comment (SammyO) [inline changelog]: This doesn't belong here. MSAL changelog is for changes that happened in MSAL.

File: msal/build.gradle [hunk 4]
```diff
@@ -188,12 +188,12 @@ task sourcesJar(type: Jar) {
...
 // In dev, we want to keep the dependencies (common4j, common) to 1.0.+ to be able to be consumed by daily dev pipeline.
 // In release/*, we change these to specific versions being consumed.
-def commonVersion = "1.0.+"
+def commonVersion = "17.6.0-RC1"
 if (project.hasProperty("distCommonVersion")) {
     commonVersion = project.distCommonVersion
 }
 // Used for testfixtures
-def common4jVersion = "1.0.+"
+def common4jVersion = "14.6.0-RC1"
```
Comment (SammyO) [inline msal/build.gradle]: I don't think this is right either? Neither the above.

File: msal/versioning/version.properties [hunk 5]
```diff
@@ -1,3 +1,3 @@
 #Wed Aug 01 15:24:11 PDT 2018
-versionName=5.4.0
+versionName=5.4.1-RC1
```
Comment (SammyO) [inline msal/versioning/version.properties]: This is not right either.

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailAttributeSignUpFragment.kt [hunk 6]
```diff
@@ -132,7 +132,7 @@ class EmailAttributeSignUpFragment : Fragment() {
 
                 val actionResult = authClient.signUp(
                     username = email,
-                    password = password,
+//                    password = password,
```
Comment (SammyO) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailAttributeSignUpFragment.kt]: Please remove commented out code.

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt [hunk 7]
```diff
@@ -120,7 +120,8 @@ class EmailPasswordSignInSignUpFragment : Fragment() {
 
                 val actionResult = authClient.signIn(
                     username = email,
-                    password = password
+                    password = password,
+                    scopes = listOf("User.Read")
```
Comment (SammyO) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt:124]: I'm not against it, but is it necessary?

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt [hunk 8]
```diff
@@ -279,6 +283,8 @@ class EmailPasswordSignInSignUpFragment : Fragment() {
                 binding.resultAccessToken.text =
                     getString(R.string.result_access_token_text) + accessToken
 
+                Log.d("AccessToken", accessToken)
```
Comment (SammyO) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt]: Why is this needed? If not, let's not log this. This is sensitive data and we should be setting an example.

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt [hunk 9]
```diff
@@ -112,7 +114,8 @@ class EmailSignInSignUpFragment : Fragment() {
                 val email = binding.emailText.text.toString()
 
                 val actionResult = authClient.signIn(
-                    username = email
+                    username = email,
+                    scopes = listOf("User.Read")
```
Comment (SammyO) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt:118]: Same here.

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt [hunk 10]
```diff
@@ -250,6 +253,8 @@ class EmailSignInSignUpFragment : Fragment() {
                 binding.resultAccessToken.text =
                     getString(R.string.result_access_token_text) + accessToken
 
+                Log.d("AccessToken", accessToken)
```
Comment (SammyO) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt]: Same here

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt (hunk 7 above)
Comment (Yuki-YuXin) [reply testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt:124]: [use case 2.2.6] Ability to provide scope to control auth strength of the token
 In this test case, we need to modify the code, providing  scopes = listOf("User.Read") . If it had been there all along, the testers wouldn't have needed any modifications

File: msal/build.gradle (hunk 4 above)
Comment (Yuki-YuXin) [reply msal/build.gradle]: That might because I merged release branch into the test app branch in order to have both test environment. I will check the content in the dev branch and revert it back.

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt (hunk 7 above)
Comment (SammyO) [reply testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt:124]: Got it.

**Review by SammyO — APPROVED** [review]
